pkg/sql/parser/sql.y

```diff
@@ -4737,12 +4737,13 @@ constraint_elem:
       },
     }
   }
-| PRIMARY KEY '(' index_params ')' opt_hash_sharded
+| PRIMARY KEY '(' index_params ')' opt_interleave opt_hash_sharded
```

Comment:
I forgot to mention but you should have a parse test for this. Other than this lgtm